client/my-sites/upgrades/plans/index.jsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import localize from 'lib/mixins/i18n/localize';
+import { getPlans } from 'state/plans/selectors';
+import { getPlansBySite } from 'state/sites/plans/selectors';
+import PlanList from 'components/plans/plan-list' ;
+import Gridicon from 'components/gridicon';
+import HeaderCake from 'components/header-cake';
+
+const Plans = ( { translate, domain, sites, plans, sitePlans, onGoBack } ) => {
+	const site = sites.getSelectedSite();
+	const comparePlansUrl = '/domains/add/' + domain + '/plans/compare/' + site.slug;
+
+	return (
+		<div className="plans has-sidebar">
+			<HeaderCake onClick={ onGoBack }>
+				{ translate( 'Register %(domain)s', { args: { domain } } ) }
+			</HeaderCake>
+
+			<PlanList { ...{ site, plans, sitePlans, comparePlansUrl } }
+				hideFreePlan={ true }
+				isInSignup={ true } />
+			<a href={ comparePlansUrl } className="plans-step__compare-plans-link">
+				<Gridicon icon="clipboard" size={ 18 } />
+				{ translate( 'Compare Plans' ) }
+			</a>
+		</div>
+	);
+};
```

Comment:
This is fine, but be aware that there are still some oddities with single-function React components, such as always re-rendering and not working with the React hot loader